File: Go/condition/condition.go
package condition

// สร้าง function เพื่อคำนวณเกรด
// ใช้ if, else if โดยที่เกรด A ต้อง 80 ขึ้นไป, เกรด B ได้ระหว่าง 70-79, ต่ำกว่านั้น F
func CalculateGrade(score int) rune {
	var grade rune

	if score >= 80 {
		grade = 'A'
	} else if score > 70 && score < 80 {
		grade = 'B'
	} else {
		grade = 'F'
	}
	return grade
}

// แสดงเกรด โดยใช้ switch case
// รับอักขระของเกรดเข้ามา และ return ข้อความออกไป
func ShowGrade(grade rune) string {
	switch grade {
	case 'A':
		return "Your grade is A"
	case 'B':
		return "Your grade is B"
	default:
		return "Your grade is F"
	}

}

// เขียน if-else ในรูปแบบ short-statement
// เป็น function ที่ใช้เพื่อเปิดปิดอุปกรณ์
func IsActive(num int) string {
	if num == 1 { // 1 แทนการเปิดเครื่อง
		return "Device is On" //ถ้าเงื่อนไขเป็จริง จะเข้าการทำงานใน block นี้
	}
	return "Device is Off" // ถ้าเป็น false จะทำงานตรงนี้
}

File: Java/condition/condition.java
package condition;

public class condition {

  public static void main(String[] args) {}

  //ใช้ if else เพื่อคำนวณคะแนนออกมาเป็นเกรด เช่น A, B, F
  public static char gradeCal(int score) {
    char grade;

    if (score >= 80) {
      grade = 'A';
    } else if (score > 70 && score < 80) {
      grade = 'B';
    } else {
      grade = 'F';
    }
    return grade;
  }

  //variable = (condition) ? expressionTrue :  expressionFalse;
  // if ในรูปแบบ short statement
  public static void isStatus(int status) {
    String result = (status == 0)
      ? "device turn off!!\n"
      : "device turn on!!\n";
    System.out.println(result);
  }

  //ใช้ switch case เพื่อหาเกรดที่ตรงตามเคส และแสดงผลออกมา
  public static void showGrade(char grade) {
    switch (grade) {
      case 'A':
        System.out.println("Your grade is A\n");
        break;
      case 'B':
        System.out.println("Your grade is B\n");
        break;
      case 'F':
        System.out.println("Your grade is F\n");
        break;
      default:
        System.out.println("Invalid grade\n");
        break;
    }
  }
}
